kotlinx-coroutines-core/common/src/channels/Produce.kt

```diff
@@ -141,3 +153,20 @@ internal open class ProducerCoroutine<E>(
         if (!processed && !handled) handleCoroutineException(context, cause)
     }
 }
+
+private class DropWhileBusyProducerCoroutine<E>(
+    parentContext: CoroutineContext,
+    channel: Channel<E>
+) : ProducerCoroutine<E>(parentContext, channel) {
+
+    private val hasSentFirstValue = atomic(false)
+
+    override suspend fun send(element: E) {
+        if (hasSentFirstValue.value) {
+            _channel.offer(element)
+        } else {
+            _channel.send(element)
+            hasSentFirstValue.getAndSet(true)
+        }
```

Comment:
There's still a race here. Two threads calling `send` concurrently will both do `send`. You should do `if (hasSentFirstValue.getAndSet(true)) _channel.offer(element) else _channel.send(element)`